### PR TITLE
[TRAFODION-2214]Supporting OLAP function LEAD and LAG, check compGeneral/TEST006.

### PR DESCRIPTION
**Pull request by HowardQin**

### CODE REVIEW

**Comment by svarnau** [comment]
Is there a JIRA?  Please edit the PR title to have [TRAFODION-XXX] at the beginning, so PR activity is logged there.

**Comment by traflm** [comment]
I created TRAFODION-2214 to track this RP.

File: core/sql/parser/sqlparser.y [hunk 1]
```diff
@@ -780,6 +781,7 @@ static void enableMakeQuotedStringISO88591Mechanism()
 %token <tokval> TOK_LAST_SYSKEY
 %token <tokval> TOK_LASTSYSKEY
 %token <tokval> TOK_LCASE               /* ODBC extension   */
+%token <tokval> TOK_LEAD                /* LEAD OLAP Function */
```
Comment (traflm) [inline core/sql/parser/sqlparser.y]: It is better to add TOK_LEAD and TOK_LAG into the nonreserved_func_word production as well.

**Comment by traflm** [comment]
looks very good for me, except a minor suggestion. That can be fixed in a later check in from my point of view. So it is good to merge from my part.
And as Steve suggested, update the title with the new JIRA , so it can be checked.

This is a very cool new feature.

**Comment by traflm** [comment]
jenkins, extra tests

**Comment by sureshsubbiah** [comment]
+1. Nicely done.

**Pull request merged**